From 183eaa304570330f1f76f6f5ffbbb3371582aec7 Mon Sep 17 00:00:00 2001
From: Cyril Brulebois <kibi@debian.org>
Date: Tue, 30 Dec 2014 14:40:50 +0100
Subject: [PATCH] Fix the empty sources.list bug with foreign architectures
 (Closes: #732255, #773867).

Update setup_apt_sources to look at USE_COMPONENTS if COMPONENTS is
empty, so that some iteration over defined components happens.
---
 debian/changelog | 3 +++
 functions        | 4 +++-
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/debian/changelog b/debian/changelog
index 925c8c6..75dd347 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -3,6 +3,9 @@ debootstrap (1.0.67) UNRELEASED; urgency=medium
   * Apply patch by Jérémy Bobbio to support reproducible builds: specify
     a modification time on the tar side, and add the -n option to gzip
     (Closes: #774069). Thanks, Jérémy!
+  * Update setup_apt_sources to look at USE_COMPONENTS if COMPONENTS is
+    empty, fixing the empty sources.list bug with foreign architectures
+    (Closes: #732255, #773867).
 
  -- Cyril Brulebois <kibi@debian.org>  Mon, 29 Dec 2014 15:01:57 +0100
 
diff --git a/functions b/functions
index 674d649..bf999f1 100644
--- a/functions
+++ b/functions
@@ -949,9 +949,11 @@ mv_invalid_to () {
 
 setup_apt_sources () {
 	mkdir -p "$TARGET/etc/apt"
+	# Cope with the foreign case where COMPONENTS would be empty:
+	sources_components="${COMPONENTS:-$USE_COMPONENTS}"
 	for m in "$@"; do
 		local cs=""
-		for c in $COMPONENTS; do
+		for c in $sources_components; do
 			local path="dists/$SUITE/$c/binary-$ARCH/Packages"
 			local pkgdest="$TARGET/$($DLDEST pkg "$SUITE" "$c" "$ARCH" "$m" "$path")"
 			if [ -e "$pkgdest" ]; then cs="$cs $c"; fi
-- 
2.1.4

